doc/crypto/OPENSSL_Applink.pod

```diff
@@ -19,6 +21,80 @@ compiler of their choice and link it into the target application.
 The referred module is available as F<applink.c>, located alongside
 the public header files (only on the platforms where applicable).
 
+B<OPENSSL_SetApplink> is allows developers to control the search for the
+B<OPENSSL_Applink> function. 
+
+=head1 NOTES
+
+The B<Applink> system is designed to solve Microsoft Visual C Runtime (MSVRT)
+interoperability issues when working with B<BIO_s_fd> and B<BIO_s_file>. If your
+application does not use any of these functions, you can disregard this system.
+
+The B<Applink> infrastructure was created to overcome the problems of mixing
+threading models. During compilation, the MSVCRT assumes the layout of B<FILE>
```

Comment:
Mixing threading models is only one particular case. I mean issue is wider than that. Intention was to facilitate mixing different compilers and/or versions, or even compiler flags. I mean say you want to compiler your application with debugging flags. Without applink you'd have to recompile even OpenSSL dlls with same flags... Or consider I've got dll from somebody else, without applink you'd have to figure out which compiler version was used and use same. With applink you can use any compiler, not even from same vendor. Once again, *if* you use interfaces that actually rely on applink. As correctly pointed out those accepting FILE * and Unix-like fd from application. And once again, recommendation is avoid those in new application code.